src/nfa/nfa.cc

```diff
@@ -320,16 +320,47 @@ void Delta::add(State state_from, Symbol symbol, State state_to) {
     } else {
         const auto symbol_transitions{ state_transitions.find(Move{ symbol }) };
         if (symbol_transitions != state_transitions.end()) {
-            // Add transition with symbolOnTransition already used on transitions from stateFrom.
+            // Add transition with symbol already used on transitions from state_from.
             symbol_transitions->insert(state_to);
         } else {
-            // Add transition to a new Move struct with symbolOnTransition yet unused on transitions from stateFrom.
+            // Add transition to a new Move struct with symbol yet unused on transitions from state_from.
             const Move new_symbol_transitions{ symbol, state_to };
             state_transitions.insert(new_symbol_transitions);
         }
     }
 }
 
+void Delta::add(const State state_from, const Symbol symbol, const StateSet& states) {
+    if(states.empty()) {
+        return;
+    }
+
+    const State max_state{ std::max(state_from, states.back()) };
+    if (max_state >= posts.size()) {
+        reserve_on_insert(posts, max_state + 1);
+        posts.resize(max_state + 1);
+    }
+
+    Post& state_transitions{ posts[state_from] };
+
+    if (state_transitions.empty()) {
+        state_transitions.insert({ symbol, states });
+    } else if (state_transitions.back().symbol < symbol) {
+        state_transitions.insert({ symbol, states });
```

Comment:
The constructor of Move should probably be made explicity so that we have to write Move{...